pkg/madmin/remote-target-commands.go

```diff
@@ -118,20 +118,20 @@ func (t *BucketTarget) Clone() BucketTarget {
 }
 
 // URL returns target url
-func (t BucketTarget) URL() string {
+func (t BucketTarget) URL() *url.URL {
 	scheme := "http"
 	if t.Secure {
 		scheme = "https"
 	}
 	urlStr := fmt.Sprintf("%s://%s", scheme, t.Endpoint)
 	u, err := url.Parse(urlStr)
 	if err != nil {
-		return urlStr
+		return nil
```

Comment:
```suggestion
		return nil
```

This is going to become silent? :-) it is better to construct the URL yourself 

```
&url.URL{
  Host: t.Endpoint,
  Scheme: scheme,
}
```

At this point AFAIK there should be no room for an error to occur with url.Parse()